app/controllers/inventories_controller.rb

```diff
@@ -1,37 +1,29 @@
 class InventoriesController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_model, only: %i[show destroy restore]
+  before_action :set_model, only: %i[destroy restore]
+  before_action :set_data, only: %i[new create]
 
   # GET /inventories
   # GET /inventories.json
   def index
-    @warehouse = Phone.joins(:inventories, :memory, :model).select(:model_id, :name, 'SUM(quantity) as total_quantity', 'MAX(price) as max_price', 'MIN(price) as min_price', 'MAX(amount) as max_amount', 'MIN(amount) as min_amount').group(:model_id) # rubocop:disable Layout/LineLength
+    @warehouse = Phone.joins(:memory).joins("INNER JOIN inventories ON inventories.phone_id = phones.id AND inventories.discarded_at IS NULL").joins("INNER JOIN models ON models.id = phones.model_id AND models.discarded_at IS NULL").select(:model_id, :name, 'SUM(quantity) as total_quantity', 'MAX(price) as max_price', 'MIN(price) as min_price', 'MAX(amount) as max_amount', 'MIN(amount) as min_amount').group(:model_id) # rubocop:disable Layout/LineLength
```

Comment:
Actually I chose using this query to avoid N+1, cause I'm using [datatable](https://datatables.net/) to perform pagination with JavaScript (not with kaminari gem), including search and sort. It required all the record I want to show up at once, so if I use your solution, I may change quite big (using pagination at server side, re-designing table on view, using search with table). At first, I thought about your solution but looking the pros of datatable made me chose this.